File: SDK-sources/OutbrainSDK/Models/Content/OBImageInfo.m
//
//  OBImageInfo.m
//  OutbrainSDK
//
//  Created by Oded Regev on 12/12/13.
//  Copyright (c) 2013 Outbrain inc. All rights reserved.
//

#import "OBImageInfo.h"
#import "OBContent_Private.h"


@implementation OBImageInfo

+ (instancetype)contentWithPayload:(NSDictionary *)payload
{
    OBImageInfo * imageInfo = [super contentWithPayload:payload];
    
    if (payload[@"imageImpressionType"])
    {
        imageInfo.isGif = [payload[@"imageImpressionType"] isEqualToString:@"DOCUMENT_ANIMATED_IMAGE"];
    }

    return imageInfo;
}

+ (NSArray *)requiredKeys
{
    return @[@"url"];
}

+ (NSDictionary *)propertiesMap
{
    return @{
             @"width":  @"width",
             @"height": @"height",
             @"url":    @"url"
             };
}

+ (Class)propertyClassForKey:(NSString *)key
{
    if ([key isEqualToString:@"url"]) {
        return [NSURL class];
    }
    
    return [super propertyClassForKey:key];
}


@end

File: SwiftSDK-sources/OutbrainSDK/Model/OBImageInfo.swift
//
//  OBRecImage.swift
//  OutbrainSDK
//
//  Created by Dror Seltzer on 20/06/2023.
//

import Foundation

@objc public class OBImageInfo: NSObject {
    
    @objc public let width: Int
    @objc public let height: Int
    @objc public let url: URL?
    
    init(width: Int, height: Int, url: URL?) {
        self.width = width
        self.height = height
        self.url = url
    }
}
